chapters/VK_EXT_debug_utils.adoc

```diff
@@ -28,8 +28,18 @@ callback to capture debug messages from a variety of Vulkan components.
 
 It can be useful for an application to provide its own content relative to a
 specific Vulkan object.
+
 The following commands allow application developers to associate
 user-defined information with Vulkan objects.
+These commands are device-level commands but they may: reference instance-level
+objects with a few restrictions.
+If the object is an instance-level object (such as a slink:VkInstance) or
+physical-device-level object (such as a slink:VkPhysicalDevice or
+slink:VkDisplayKHR), then the lifetime of the debug annotated data provided to
+one or more of the following commands is restricted to the lifetime of the
+associated slink:VkDevice object used as the parameter.
+Therefore, the data is only valid as long as the slink:VkDevice object is valid,
+or until a corresponding call is made to change the data for that object.
```

Comment:
This is an interesting restriction.  If an application sets a debug object name on a VkInstance or VkPhysicalDevice from two different VKDevices, what happens?  Is it up to the implementation to pick the name for a debug message?

Reply:
This is a device command, so only the VkPhysicalDevice associated with the device used to call would be used.  At least that's what I'm suggesting.

Reply:
I can tell you from the loader's implementation that the restriction of only being able to name the VkPhysicalDevice that created the VkDevice is essential, since its not practical to 'get' the VkPhysicalDevice associated with non-created VkDevices. 
Since the desktop loader 'wrap's the VkPhysicalDevice handles, the loader must intercept and modify the VkPhysicalDevice parameters.

Reply:
Okay, what if two different Devices created from two different PhysicalDevices are called to set the debug name for an Instance?  It's unlikely, but e.g.
```
vkCreateDevice(physical_device1, ..., &device1);
vkCreateDevice(physical_device2, ..., &device2);
vkSetDebugObjectNameEXT(device1, {VK_STRUCTURE_TYPE_DEBUG_UTILS_OBJECT_NAME_INFO_EXT,..., VK_OBJECT_TYPE_INSTANCE, instance, "an instance"} );
vkSetDebugObjectNameEXT(device2, {VK_STRUCTURE_TYPE_DEBUG_UTILS_OBJECT_NAME_INFO_EXT,..., VK_OBJECT_TYPE_INSTANCE, instance, "my instance"} );
```

Reply:
The loader doesn't care in this case, it just forwards the `vkSetDebugObjectNameEXT` to whoever does. That said, I believe this is just a case of determining whether VkInstance should *always* have a single name, or if there should be a special case here.
I am for a single name, so the above code would result in VkInstance->"my instance".

Reply:
That is ambiguous with the proposed wording - if device1 sets the instance's name, then device2 sets the name, then device2 is destroyed, the implication is that the instance's name either has to be cleared or reverts to the name set through device1.

I think maybe it's easier to omit the bit about the lifetime of the set name to the VkDevice's lifetime unless that's how implementations already work.

Maybe something about the name most recently set on the instance or the physical device sticks instead.

(The implication from all of this to me is that SetDebugObjectNameExt probably either should have been on Instance or current use to set the name on a Device's parent (e.g. vkquake does it on PhysicalDevice) should become disallowed / invalid usage.  Future maintenance extension or vulkan revision, I guess)

Reply:
Okay after talking with brad, I think specifying the lifetime here causes more ambiguity and implications for implementations. The issue is that a debug utils implementation would have to 'track' *who* set the name and keep a list of names associated with the device who set it. Whereas the lack of lifetime specification would allow implementations to have a single name for the handle, which should cover all reasonable use cases.

Reply:
But this doesn't solve the problem.  It depends on where the implementation stores the name.  If the name is associated in the device, it will go away and not be avaliable afterwords.  But, if it associates it with the object directly, then it will stick.  Since we weren't clear originally, I feel like we almost need to support both. 

So are we saying we just strip out the lifetime comment then and leave it:

```
These commands are device-level commands but they may: reference instance-level
objects with a few restrictions.
```

But what about the case where the implementations stores the names with the device.  Shouldn't we have a statement akin to `this data may go away on the destruction of the device`?

That's my concern.  If we don't clarify when the data is useful, then people will try to use it when it's not valid.

Reply:
You didn't address the issue I brought up above; if names are set on a `VkInstance` or `VkPhysicalDevice` through two different `VkDevice`s, then one `VkDevice` is deleted, which name persists?  I'm trying to get to the fundamental question about to which object the name is attached.

I think the common use case you mentioned to me in person yesterday is that two `VkDevice`s are created, one for graphics use and the other to set debug names.  That implies the name on the `VkInstance` is expected to be valid outside the scope of the debug name `VkDevice`.  So it sounds to me like the name has to be attached to `VkInstance` and `VkPhysicalDevice` with no association with a child `VkDevice`.

We discussed in person yesterday that in the case a layer or implementation wants to print debug information about the `VkInstance` or `VkPhysicalDevice` that we could consider the name only valid in the context of calling a device function.  But that doesn't handle the common use case above.  Now I think the extension should just say "it sets the name, replacing any previously set name".  Then there's no ambiguity about lifetime if a name is set through two separate `VkDevice`s or if it's set through one `VkDevice` but a layer or implementation wants to print information about the `VkInstance` in a function on call on a different `VkDevice`.

So then the language could look like this if you just want to get this done and move on:

```
These commands are device-level commands but they may: reference instance-level
objects.  A name set on an instance-level object replaces any name previously set on
that object.
```

I think to do do this well we should probably just survey what apps and layers and implementations do and then retconn the extension to cover the most common case.  That's probably too much trouble to cover a small corner case that everyone is already working around.

Reply:
Perhaps:

```
For example, all object data may be stored in such a way as to relate it to the VkDevice handle and may not be accessible after the destruction of the VkDevice, even if the data is associated with instance or physical device-level objects.
```

Reply:
Does that mean you can't set instance or physical device names through one device and have them show up in debug prints in another device?

Reply:
I think that's possible.  I'm trying to be vague enough to allow it since implementations of this have been out in the wild for 5 years or so now and I don't want to require new behavior, just clarify existing behavior.